src/coreclr/jit/compiler.cpp

```diff
@@ -560,6 +560,23 @@ bool Compiler::isTrivialPointerSizedStruct(CORINFO_CLASS_HANDLE clsHnd) const
 }
 #endif // TARGET_X86
 
+bool Compiler::isNativePrimitiveStructType(CORINFO_CLASS_HANDLE clsHnd)
```

Comment:
Yep! The only place where "is this type a struct" matters in calling conventions is in `getReturnTypeForStruct`, so that's the only place we need to touch today.

And since `CLong` and `CULong` are 32-bits on x86 Windows and `NFloat` is technically an HFA, we won't have to do any additional work for these types specifically if we add support for __fastcall or __vectorcall calling conventions in the future.